pkg/kubectl/run.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+)
+
+type BasicReplicationController struct{}
+
+func (BasicReplicationController) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"name", true},
+		{"replicas", true},
+		{"image", true},
+		{"command", false},
+	}
+}
+
+func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+	labelString, found := params["labels"]
+	var labels map[string]string
+	if found && len(labelString) > 0 {
+		labels = ParseLabels(labelString)
+	} else {
+		labels = map[string]string{
+			"name": params["name"],
+		}
+	}
+	count, err := strconv.Atoi(params["replicas"])
+	if err != nil {
+		return nil, err
+	}
+	controller := api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name:   params["name"],
+			Labels: labels,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: count,
+			Selector: labels,
+			Template: &api.PodTemplateSpec{
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:    params["name"],
+							Image:   params["image"],
+							Command: ParseCommand(params["command"]),
+						},
+					},
+				},
+			},
+		},
+	}
+	return &controller, nil
+}
+
+func ParseLabels(labelString string) map[string]string {
+	if len(labelString) == 0 {
+		return nil
+	}
+	labels := map[string]string{}
+	labelSpecs := strings.Split(labelString, ",")
+	for ix := range labelSpecs {
+		labelSpec := strings.Split(labelSpecs[ix], "=")
+		if len(labelSpec) != 2 {
+			glog.Errorf("unexpected label spec: %s", labelSpecs[ix])
+			continue
+		}
+		labels[labelSpec[0]] = labelSpec[1]
+	}
+	return labels
+}
+
+func ParseCommand(cmd string) []string {
+	if len(cmd) == 0 {
+		return nil
+	}
+	return strings.Split(cmd, " ")
```

Comment:
This seems totally dangerous. What if there are quoted or escaped spaces in the command? We need a command string in the API, as per #2316. Could we just not expose this until we have that? It seems likely to lead to user confusion.